doc/mgmt/SONiC_YANG_Model_Guidelines.md

```diff
@@ -622,7 +623,180 @@ module sonic-port {
 ```
 
 
+## SONiC YANG Extensions
 
+#### 1. db-name: When SONiC YANG to be used as Northbound YANG, application may need to get data from DBs other than CONFIG_DB such as STATE_DB or APPL_DB. Use "db-name" extension to specify the DB where this table is defined.
+
+#### 2. key-delim: When "db-name" extension is used to specify DB type other than CONFIG_DB, use "key-delim" extension to specify the key delimiter used for this DB type.
+
+Example:
+```
+container RADIUS_SERVER_STATS {
+    sonic-ext:db-name "COUNTERS_DB";
+    sonic-ext:key-delim ":";
+    config false;
+    list RADIUS_SERVER_STATS_LIST {
+    ....
+    ....
+    }
+}
+```
+
+#### 3. key-pattern: Use this extension when table key follows special pattern.
+
+Example:
+##### ABNF:
+```
+    "QUEUE": {
+        "Ethernet0,Ethernet4,Ethernet8,Ethernet12,Ethernet16,Ethernet120,Ethernet124|0": {
+            "scheduler": "[SCHEDULER|scheduler.1]"
+        },
+        "Ethernet0,Ethernet4,Ethernet8,Ethernet12,Ethernet16,Ethernet100,Ethernet104,Ethernet124|3-4": {
+            "wred_profile": "[WRED_PROFILE|AZURE_LOSSLESS]",
+            "scheduler": "[SCHEDULER|scheduler.0]"
+        }
+    }
+```
+##### YANG:
+```
+container QUEUE {
+    list QUEUE_LIST {
+        key "ifname qindex";
+        sonic-ext:key-pattern "QUEUE|({ifname},)*|{qindex}";
+        
+        leaf ifname {
+            type leafref {
+                path "/prt:sonic-port/prt:PORT/prt:PORT_LIST/prt:ifname";
+            }
+        }
+        leaf qindex {
+            type string;
+        }
+        ....
+    }
+}
+```
+
+#### 4. map-leaf: Mapping tables in Redis are defined using nested 'list'. The outer 'list' is used for multiple instances of mapping. The inner 'list' is used for mapping entries for each outer list instance. Use 'map-leaf' extension to indicate nested 'list' and specify mapping between two fields which are also key fields of inner list.
```

Comment:
Is this extension supported  in the python based library  or only CVL ?

Reply:
As of now, it is supported in CVL only.